src/main/java/org/jabref/logic/importer/fetcher/DoiFetcher.java

```diff
@@ -47,8 +55,17 @@ public String getName() {
     @Override
     public Optional<BibEntry> performSearchById(String identifier) throws FetcherException {
         Optional<DOI> doi = DOI.parse(identifier);
+
         try {
             if (doi.isPresent()) {
+                Optional<BibEntry> fetchedEntry;
+
+                // mEDRA does not return a parsable bibtex string
+                if (getAgency(doi.get()).isPresent() && "medra".equalsIgnoreCase(getAgency(doi.get()).get())) {
+                    fetchedEntry = new Medra().performSearchById(identifier);
```

Comment:
Simply `return new Medra()...`  should work.

Reply:
Ok